src/App.tsx

```diff
@@ -14,11 +14,88 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGood: string[],
+};
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGood: ['Jam'],
+  };
+
+  // methods for adding & removing elements from an array return array
+  // filter,slice,map,concat
+
+  goodsInCourt = (goods: string[] | []) => {
+    const { selectedGood }: { selectedGood: string[] } = this.state;
+
+    if (goods.length === 0) {
+      return 'No goods selected';
+    }
+
+    if (goods.length === 1) {
+      return `${selectedGood[0]} is selected`;
+    }
+
+    return `${goods.slice(0, -1).join(',')} and ${goods[goods.length - 1]}`;
```

Comment:
Add one more space after comma in join method, because now your selected goods is merging)